File: backend/app/workflow/tools/think.py
from langchain_core.tools import tool


@tool
def think_tool(reflection: str) -> str:
    """Strategic reflection tool. Use to pause and reason about findings, gaps,
    or next steps. Does not perform any I/O — just records the reflection.

    Args:
        reflection: A short paragraph of reasoning about the current research state.
    """
    return f"Reflection recorded: {reflection}"

File: worker/src/tools/think.ts
import { tool } from "@langchain/core/tools";
import { z } from "zod";

export const thinkTool = tool(
  async ({ reflection }) => `Reflection recorded: ${reflection}`,
  {
    name: "think_tool",
    description: "Reflect on findings and plan next search queries.",
    schema: z.object({
      reflection: z.string().describe("Analysis of findings, gaps, and next steps."),
    }),
  },
);
